dojo/tools/dependency_track/parser.py

```diff
@@ -0,0 +1,202 @@
+                },
+                "vulnerability": {
+                    "uuid": "115b80bb-46c4-41d1-9f10-8a175d4abb46",
+                    "source": "NPM",
+                    "vulnId": "533",
+                    "title": "Regular Expression Denial of Service",
+                    "subtitle": "timespan",
+                    "severity": "LOW",
+                    "severityRank": 3,
+                    "cweId": 400,
+                    "cweName": "Uncontrolled Resource Consumption ('Resource Exhaustion')",
+                    "description": "Affected versions of `timespan`...",
+                    "recommendation": "No direct patch is available..."
+                },
+                "analysis": {
+                    "state": "NOT_SET",
+                    "isSuppressed": false
+                },
+                "matrix": "ca4f2da9-0fad-4a13-92d7-f627f3168a56:b815b581-fec1-4374-a871-68862a8f8d52:115b80bb-46c4-41d1-9f10-8a175d4abb46"
+            },
+            {
+                "component": {
+                    "uuid": "979f87f5-eaf5-4095-9d38-cde17bf9228e",
+                    "name": "uglify-js",
+                    "version": "2.4.24",
+                    "purl": "pkg:npm/uglify-js@2.4.24"
+                },
+                "vulnerability": {
+                    "uuid": "701a3953-666b-4b7a-96ca-e1e6a3e1def3",
+                    "source": "NPM",
+                    "vulnId": "48",
+                    "title": "Regular Expression Denial of Service",
+                    "subtitle": "uglify-js",
+                    "severity": "LOW",
+                    "severityRank": 3,
+                    "cweId": 400,
+                    "cweName": "Uncontrolled Resource Consumption ('Resource Exhaustion')",
+                    "description": "Versions of `uglify-js` prior to...",
+                    "recommendation": "Update to version 2.6.0 or later."
+                },
+                "analysis": {
+                    "isSuppressed": false
+                },
+                "matrix": "ca4f2da9-0fad-4a13-92d7-f627f3168a56:979f87f5-eaf5-4095-9d38-cde17bf9228e:701a3953-666b-4b7a-96ca-e1e6a3e1def3"
+            }]
+    }
+    """
+
+    def _convert_dependency_track_severity_to_dojo_severity(self, dependency_track_severity):
+        """
+        Converts a Dependency Track severity to a DefectDojo severity.
+        :param dependency_track_severity: The severity from Dependency Track
+        :return: A DefectDojo severity if a mapping can be found; otherwise a null value is returned
+        """
+        severity = dependency_track_severity.lower()
+        if severity == "critical":
+            return "Critical"
+        elif severity == "high":
+            return "High"
+        elif severity == "medium":
+            return "Medium"
+        elif severity == "low":
+            return "Low"
+        elif severity.startswith("info"):
+            return "Informational"
+        else:
```

Comment:
Dependency-Track also supports 'UNASSIGNED' as a severity. 
https://github.com/DependencyTrack/dependency-track/blob/master/src/main/java/org/dependencytrack/model/Severity.java

This typically occurs when there's an incomplete CVE and OSS Index (or another source) picked it up. It's usually a confirmed vulnerability, but the severity is unknown as it was never formally rated.

Reply:
I believe DefectDojo only has [Critical, High, Medium, Low, Informational] so we would need to map to one of those. Here is where I default to a Critical severity if we were unable to map a Dependency Track severity to a DefectDojo severity: https://github.com/DefectDojo/django-DefectDojo/pull/1561/files#diff-0c6cbdfc684bc0d0f4300320b93fcadcR151